### PR TITLE
a11y fixes for v7

### PR DESCRIPTION
**Pull request by hannahkowen**
Problem:
1.	The full date of a day cell in the calendar day picker grid is not announced with NVDA, just the number of the day (e.g. 11 instead of Wednesday November 11th, 2019)
2.	The month name is not announced when changing between months

Solution:
1.	Some screen readers (NVDA) do not read ariaLabels directly on `<div>`s. 
a.	Place the visible text in a span hidden from screen readers
b.	have the screen readers read from a separate span that contains the ariaLabels.
c.	Add css class to hide only the screen reader span from display.
2.	Add aria-live=”polite” to caption so that it reads the month when changing between months.

### CODE REVIEW

File: src/Caption.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default class Caption extends Component {
       onClick,
     } = this.props;
     return (
-      <div className={classNames.caption} role="heading">
+      <div className={classNames.caption} role="heading" aria-live="polite">
```
Comment (gpbl) [inline src/Caption.js:54]: @hannahkowen this is great thanks!

File: src/Day.js [hunk 2]
```diff
@@ -150,7 +149,8 @@ export default class Day extends Component {
         onTouchStart={handleEvent(onTouchStart, day, modifiers)}
         onFocus={handleEvent(onFocus, day, modifiers)}
       >
-        {children}
+        <span className="DayPicker-ScreenReaderOnly">{ariaLabel}</span>
+        <span aria-hidden="true">{children}</span>
```
Comment (gpbl) [inline src/Day.js:153]: I fear this line will be a breaking change for many. Could we just add `aria-live="polite"` and the "screen-reader-only part – or are these changes related?

File: src/Day.js [hunk 3]
```diff
@@ -150,7 +149,8 @@ export default class Day extends Component {
         onTouchStart={handleEvent(onTouchStart, day, modifiers)}
         onFocus={handleEvent(onFocus, day, modifiers)}
       >
-        {children}
+        <span className="DayPicker-ScreenReaderOnly">{ariaLabel}</span>
```
Comment (gpbl) [inline src/Day.js:152]: User can set their own [classNames](http://react-day-picker.js.org/api/DayPicker#classNames): this className should be added to the list.

File: test/daypickerinput/rendering.js [hunk 4]
```diff
@@ -119,7 +119,9 @@ describe('DayPickerInput', () => {
       wrapper.instance().showDayPicker();
       wrapper.update();
       expect(wrapper.find('.DayPicker-Day--selected')).toHaveLength(1);
-      expect(wrapper.find('.DayPicker-Day--selected')).toHaveText('15');
+      expect(wrapper.find('.DayPicker-Day--selected')).toHaveText(
+        'Fri Dec 15 201715'
```
Comment (gpbl) [inline test/daypickerinput/rendering.js:123]: This change is unexpected :)

File: src/Day.js (hunk 2 above)
Comment (trichards-shedd) [reply src/Day.js:153]: hey, friend! I'm one of Hannah's coworkers helping get this PR up to snuff. Is the issue that `Day`'s `children` would be pushed down a level in the DOM tree? We placed `children` outside of the span tag again with no poor effects from the screenreader's perspective, if that's acceptable.

File: src/Day.js (hunk 3 above)
Comment (trichards-shedd) [reply src/Day.js:152]: this className is always required to hide this from a typical user's view. are you asking that a key on the `classNames` object be added for the screenreader text?

File: test/daypickerinput/rendering.js (hunk 4 above)
Comment (trichards-shedd) [reply test/daypickerinput/rendering.js:123]: unfortunately, no matter how hard we hide it, the text the screenreader reads is still present. if they were two separate spans, we could select into them and verify each's content. could also use `toIncludeText` instead but that feels flakier :^(

File: src/Day.js (hunk 3 above)
Comment (gpbl) [reply src/Day.js:152]: Yes! Every class name used in the component can be replaced with a custom one via that `classNames` prop. We need to add it here https://github.com/gpbl/react-day-picker/blob/v7/src/classNames.js.

File: src/Day.js (hunk 2 above)
Comment (gpbl) [reply src/Day.js:153]: My problem is that I cannot add breaking changes to v7, as i have no time left :)

Could you open an issue describing better the problem?

My concern is that this change may break other DayPicker implementations: e.g. what happens to localized calendars? Must the existing DayPickers add an `ariaLabel` after this change?

File: test/daypickerinput/rendering.js (hunk 4 above)
Comment (gpbl) [reply test/daypickerinput/rendering.js:123]: The tests are there to not break them... if they break, then is a breaking change, and I cannot work on a major release right now :(

**Comment by gpbl** [comment]
Let's do this:

- open a PR with only the `aria-live=”polite”` change. This is not breaking so it is easy to publish.
- open a new issue describing the other problem. I don't understand, for example, if this is only for the NVDA readers. 

I'd prefer to find a solution that is not breaking. I am working on v8 so I don't have time for this. If you prefer [help me there](https://github.com/gpbl/react-day-picker/projects/7)!

Thanks!
